addons/web/controllers/main.py

```diff
@@ -1057,6 +1057,23 @@ def image(self, model, id, field, **kw):
             image_data = self.placeholder()
         headers.append(('ETag', retag))
         headers.append(('Content-Length', len(image_data)))
+
+        # Guess mime type
+        signatures = {
+            "image/jpeg": ['\xFF\xD8\xFF\xE0', '\xFF\xD8\xFF\xE2',
+                           '\xFF\xD8\xFF\xE3', '\xFF\xD8\xFF\xE1'],
+            "image/gif": ['GIF87a', 'GIF89a'],
```

Comment:
I'm not sure if the problem is uploading. See the error (can't display the image). Maybe it's because of this.